server/public/shared/driver/driver.go

```diff
@@ -35,7 +35,14 @@ func NewConnector(api plugin.Driver, isMaster bool) *Connector {
 }
 
 func (c *Connector) Connect(_ context.Context) (driver.Conn, error) {
-	connID, err := c.api.Conn(c.isMaster)
+	// The passing of an empty string is unfortunate.
+	// Because the pluginID is actually passed in the internal driver implementation.
+	// But the issue is that the internal implementation is in the app package, so we cannot
+	// import the app package inside the plugin package. Therefore, we have to stick
+	// with a common API for both the plugin and the internal consumer.
+	//
+	// But it's not a big deal, since all plugins are meant to work via the pluginapi.
```

Comment:
Discussed this internally. As long as the plugin implementation is using `driver.NewConnector()` then updating the server dependency takes care of this automatically so there's no strict requirement to use the [`pluginapi`](https://github.com/mattermost/mattermost/tree/master/server/public/pluginapi) wrapper in this case.

Reply:
> updating the server dependency

Does this PR take effect only for plugins that update their dependencies and recompile?

Reply:
(from the looks of the code, it seems like no: just wanted to make sure I understand correctly)

Reply:
> Does this PR take effect only for plugins that update their dependencies and recompile?

@lieut-data I'd expect so because the `pluginID` is injected from the client RPC side.

Reply:
https://github.com/mattermost/mattermost/blob/ca26cd3ff6fa6b9bf2ab2216a9013b5b327c0bab/server/public/plugin/client_rpc.go#L259-L262

Reply:
No, plugins don't need to do anything. They just need to run a server version where this PR exists.

Reply:
Okay now I am confused on what is compiled on plugin side and what's not. Need to take a deeper look.

Reply:
To make things clear: `hooksRPCClient` runs on server, and `hooksRPCServer` runs on the plugin. I know, very intuitive.

Reply:
Oh that makes perfect sense :sweat_smile: